org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.xtend

```diff
@@ -288,10 +287,7 @@ import static org.eclipse.xtext.diagnostics.Severity.*
 					dirtyFiles += key
 				}
 			]
-			var Buildable buildable = []
```

Comment:
@JanKoehnlein how does this actually differ? or is it somehow required to actually do something on the buildable?

Reply:
(i guess the submit)

Reply:
In my case, both lists (`deletedFiles` and `dirtyFiles`) were empty, likely because the `dirtyFile` was pruned as being open, so the empty `Buildable` does nothing. As opposed to that, `workspaceManager#didChangeFiles` calls `submit` which returns the `Buildable` that actually calls `internalBuild`.